### ISSUE TITLE
Hello, I'm trying to add your cmake toolchain file to cc65 project.

### ISSUE THREAD

**Issue by RomanKovtyukh** [fo-fo/ngin#2]
Link to pull-request: https://github.com/cc65/cc65/pull/680
Wouldn't you mind?

**Comment by fo-fo** [comment]
Yeah that's fine by me. I'd appreciate though if you could change the top lines to something like:
```
# CMake toolchain file for cc65
# by thefox <thefox@aspekt.fi>
```

If you have any problems with it let me know.

**Comment by RomanKovtyukh** [comment]
Ok. The maintainer proposed me to publish it on project's wiki instead of placing it in project tree. Now I have some troubles with it on my machine, but I don't know exactly if problem in toolchain file, so as soon as I fix my build process, I'll post your toolchain file with new first lines)

**Comment by RomanKovtyukh** [comment]
Hi. I already fixed problems I have and they were not in toolchain file. But I have a question. Why do you prevent cmake from checking compiler?
I'm trying to make cmake do his check. Now I added "CC65_HOME" environment variable to make cc65 able to find files it needs, but there is some other issues now with cc65 command line options that are not present and with some specific behavior (it creates temporary .s files in the .c source file directory).
So now I copied some cmake files from /usr/share/cmake-<version>/Modules/Compiler/: GNU{,-ASM,-C}.cmake and trying to adapt them to cc65.
Did you do something like this when you were writing toolchain-file? May be I do some strange or wrong things? What do you think about it?

**Comment by fo-fo** [comment]
> Why do you prevent cmake from checking compiler?

If I remember correctly, I think CMake was by default running tests that required running the compiled executables (it's checking the support of various compiler features). That's kind of difficult when cross-compiling (although maybe the simulator target of cc65 could be used for this -- that is, you could compile the tests for the simulator target and then run them through the `sim65` simulator).

It might be possible to make it work, I just couldn't bother with it.

But yeah I think my toolchain file was also partially based on the default compiler files from CMake.

**Comment by RomanKovtyukh** [comment]
CMake used for a lot cross compiling tasks and I don't think it runs compiled code while checking compiler if it's crosscompiling. It's probably was too long ago to think about it now, because it was at least a different version of CMake.

Anyway now I see that toolchain-file works even with some checks failed (ABI/ASM) and even more, now I see that projects for NES are not so big to use CMake at all (I don't know about projects for other platforms). But CMake today is often a path from text editors to IDE (CLion/QtCreator), so some reason to finish this work I still have)

**Comment by fo-fo** [comment]
The issue at the time could have been that the toolchain didn't tell CMake that the compiler is a cross-compiler. But I really don't remember well why I did what I did, it's been a long time.

I'm not aware of many projects that use CMake with cc65 so far. But it's probably partially because not many people are aware that it's possible. Then again, many people are content with simple build scripts or makefiles.
